File: Problemas/cpp/functors/java/MainFunctor.java
class MainFunctor
{
	public static void main(String[] argv)
	{
		System.out.println("sqr()");
		Functors.plot(new Functors.Sqr());
		System.out.println();

		System.out.println("mul_by(3)");
		Functors.plot(new Functors.MulBy(3));
		System.out.println();

		System.out.println("round()");
		Functors.plot(new Functors.Round());
		System.out.println();
	}
}

File: Problemas/cpp/functors/c++/main-functor.cc
#include <iostream>
#include <iomanip>
#include "functor.h"

using namespace std;

int main(int argc, char** argv)
{
	cout << "sqr()" << endl;
	fn::plot(fn::sqr());
	cout << endl;

	cout << "mul_by(3)" << endl;
	fn::plot(fn::mul_by(3));
	cout << endl;

	cout << "round()" << endl;
	fn::plot(fn::round());
	cout << endl;

	return 0;
}


